### DISCUSSION TITLE
multi valued flags

### DISCUSSION THREAD

**Discussion by easyrider14** [bashly-framework/bashly#179] [Q&A]
Hi

Just discovered your tool, and it's very useful
I've got a question though : is it possible to define a multi valued flag ?

example :

myCmd action --param value1 --param value2

thanks

**Comment by DannyBen** [comment]
Well, no. There is no support for this now.

Personally, I always found this mechanic a little niche and awkward. From human interaction perspective, it makes more sense to me to say `--param value1,value2` than `--param value1 --param value2`.

On top of that, a multi value flag means its value is always an array - which is not a first class citizen in bashland.

That said - it is an interesting suggestion. I will see if there is an easy way to implement such a feature.

**Comment by DannyBen** [comment]
I actually believe I can implement this. It can also allow using repeatable flags for flags without arguments, for example, to provide a higher level of verbosity with `-vvv`.

The primary caveat, is that since bashly is already using an `$args` array for the input arguments, and since bash does not support nested arrays, it means that the value of repeatable flags will be a space dlimited string that needs to be converted to array in your code.

**Reply by DannyBen** [reply]
Implementation is in https://github.com/DannyBen/bashly/pull/180

**Reply by easyrider14** [reply]
thanks, that was fast
testing right away

**Reply by DannyBen** [reply]
we aim to please :)
